File: Scala Kompics/src/se/sics/kompics/scala/performancetest/kompics/kvstore/KeyValueStore.scala
package se.sics.kompics.scala.performancetest.kompics.kvstore

import se.sics.kompics.PortType
import se.sics.kompics.Event

case class KVSPut(key: String, value: Any) extends Event
case class KVSGet(key: String) extends Event
case class KVSPutReply(key: String) extends Event
case class KVSGetReply(key: String, value: Any) extends Event

object KeyValueStore extends PortType {
	PortType.preloadInstance(this);
	request(classOf[KVSPut]);
	request(classOf[KVSGet]);
	indication(classOf[KVSPutReply]);
	indication(classOf[KVSGetReply]);
}

File: Java Kompics/src/se/kth/ii2202/experiments/kvstore/KeyValueStore.java
package se.kth.ii2202.experiments.kvstore;

import se.sics.kompics.PortType;

/**
 *
 * @author carbone
 */
public class KeyValueStore extends PortType{
    
    {
        request(KVSPut.class);
        request(KVSGet.class);
        indication(KVSPutReply.class);
        indication(KVSGetReply.class);
    }
    
}
